ballerina/tests/test.bal

```diff
@@ -84,6 +84,26 @@ function testCreateObject() {
     }
 }
 
+@test:Config {
+    dependsOn: [testCreateBucket]
+}
+function testCreateObjectWithMetadata() {
+    log:printInfo("amazonS3Client->createObject()");
+    map<string> metadata = {
+        "Description" : "This is a text file",
+        "Language" : "English"
+    };
+    Client|error amazonS3Client = new(amazonS3Config);
+    if (amazonS3Client is Client) {
+        error? response = amazonS3Client->createObject(testBucketName, fileName, content, userMetadataHeaders = metadata);
+        if (response is error) {
+            test:assertFail(response.toString());
+        }
+    } else {
+        test:assertFail(amazonS3Client.toString());
+    }
+}
```

Comment:
We can improve the code by replacing if else blocks with check expressions as mentioned below

```suggestion
function testCreateObjectWithMetadata() returns error? {
    map<string> metadata = {
        "Description" : "This is a text file",
        "Language" : "English"
    };
    
    Client|error amazonS3Client = check new(amazonS3Config);
    _ = check amazonS3Client->createObject(testBucketName, fileName, content, userMetadataHeaders = metadata);
}
```